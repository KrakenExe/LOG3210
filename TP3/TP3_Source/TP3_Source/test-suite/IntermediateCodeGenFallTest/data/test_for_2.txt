Declaration
bool a;
num i;
num c;
EndDeclaration

for(c = 0; i < 56 + 78 - 6 / 2; i = i + 1) {
    a = (1 + 1) == 2;
}