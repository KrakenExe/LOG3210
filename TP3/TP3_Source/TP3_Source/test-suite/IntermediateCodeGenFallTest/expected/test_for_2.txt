c = 0
_L1
_t2 = 6 / 2
_t1 = 78 - _t2
_t0 = 56 + _t1
ifFalse i < _t0 goto _L0
_t3 = 1 + 1
ifFalse _t3 == 2 goto _L2
a = 1
goto _L1
_L2
a = 0
_t4 = i + 1
i = _t4
goto _L1
_L0
